src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -528,19 +549,32 @@ private void OnSecretsChanged(object sender, SecretsChangedEventArgs e)
 
         public async Task PurgeOldSecretsAsync(string rootScriptPath, ILogger logger)
         {
-            if (!Directory.Exists(rootScriptPath))
+            using (_metricsLogger.LatencyEvent(GetMetricEventName(MetricEventNames.SecretManagerPurgeOldSecrets)))
             {
-                return;
+                if (!Directory.Exists(rootScriptPath))
+                {
+                    return;
+                }
+
+                // Create a lookup of all potential functions (whether they're valid or not)
+                // It is important that we determine functions based on the presence of a folder,
+                // not whether we've identified a valid function from that folder. This ensures
+                // that we don't delete logs/secrets for functions that transition into/out of
+                // invalid unparsable states.
+                var currentFunctions = Directory.EnumerateDirectories(rootScriptPath).Select(p => Path.GetFileName(p)).ToList();
+
+                await _repository.PurgeOldSecretsAsync(currentFunctions, logger);
             }
+        }
 
-            // Create a lookup of all potential functions (whether they're valid or not)
-            // It is important that we determine functions based on the presence of a folder,
-            // not whether we've identified a valid function from that folder. This ensures
-            // that we don't delete logs/secrets for functions that transition into/out of
-            // invalid unparsable states.
-            var currentFunctions = Directory.EnumerateDirectories(rootScriptPath).Select(p => Path.GetFileName(p)).ToList();
+        private string GetMetricEventName(string name)
+        {
+            return string.Format(CultureInfo.InvariantCulture, name, _repositoryClassName);
```

Comment:
Will the class name be lower-cased somewhere else? I see that all our metrics are typically all lower-case, but I'm not sure if that's a requirement or not.